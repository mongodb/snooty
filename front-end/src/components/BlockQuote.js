import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';

const BlockQuote = props => {
  const { nodeData } = props;
  return (
    <section>
      {nodeData.children.map((element, index) => (
        <ComponentFactory {...props} nodeData={element} key={index} />
      ))}
    </section>
  );
};

BlockQuote.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
};

export default BlockQuote;
