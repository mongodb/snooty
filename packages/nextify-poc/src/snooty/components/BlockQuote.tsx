import React from 'react';
import PropTypes from 'prop-types';
import { BlockQuoteNode } from '../types/ast';
import ComponentFactory from './ComponentFactory';

export type BlockQuoteProps = {
  nodeData: BlockQuoteNode;
};

const BlockQuote = ({ nodeData: { children }, ...rest }: BlockQuoteProps) => (
  <blockquote>
    {children.map((element, index) => (
      <ComponentFactory {...rest} nodeData={element} key={index} />
    ))}
  </blockquote>
);

BlockQuote.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
};

export default BlockQuote;
