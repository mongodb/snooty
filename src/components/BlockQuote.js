import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';

const BlockQuote = ({ nodeData: { children }, ...rest }) => (
  <blockquote>
    {children.map((element, index) => (
      <ComponentFactory {...rest} nodeData={element} key={index} />
    ))}
  </blockquote>
);

BlockQuote.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.array.isRequired,
  }).isRequired,
};

export default BlockQuote;
