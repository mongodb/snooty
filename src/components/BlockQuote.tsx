import React from 'react';
import { BlockQuoteNode } from '../types/ast';
import ComponentFactory from './ComponentFactory';

const BlockQuote = ({ nodeData: { children }, ...rest }: { nodeData: BlockQuoteNode }) => (
  <blockquote>
    {children.map((element, index) => (
      <ComponentFactory {...rest} nodeData={element} key={index} />
    ))}
  </blockquote>
);

export default BlockQuote;
