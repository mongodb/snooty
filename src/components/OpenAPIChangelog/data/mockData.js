export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
