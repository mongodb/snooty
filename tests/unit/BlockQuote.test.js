import React from 'react';
import { shallow } from 'enzyme';
import BlockQuote from '../../src/components/BlockQuote';

// data for this component
import mockData from './data/BlockQuote.test.json';

it('renders correctly', () => {
  const tree = shallow(<BlockQuote nodeData={mockData} />);
  expect(tree).toMatchSnapshot();
});
