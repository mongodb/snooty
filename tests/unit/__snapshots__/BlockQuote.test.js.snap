// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #007CAD;
  cursor: pointer;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.emotion-1:focus {
  outline: none;
}

.emotion-2 {
  background-repeat: repeat-x;
  background-size: 2px 2px;
  background-position: center bottom;
}

[data-leafygreen-ui="anchor-container"]:hover .emotion-2 {
  background-image: linear-gradient( #E7EEEC 100%, #E7EEEC 0 );
}

[data-leafygreen-ui="anchor-container"]:focus .emotion-2 {
  background-image: linear-gradient( to right, #9DD0E7 100%, #9DD0E7 0 );
}

.emotion-2 {
  background-repeat: repeat-x;
  -webkit-background-size: 2px 2px;
  background-size: 2px 2px;
  -webkit-background-position: center bottom;
  background-position: center bottom;
}

[data-leafygreen-ui="anchor-container"]:hover .emotion-2 {
  background-image: linear-gradient(
        #E7EEEC 100%,
        #E7EEEC 0
      );
}

[data-leafygreen-ui="anchor-container"]:focus .emotion-2 {
  background-image: linear-gradient(
        to right,
        #9DD0E7 100%,
        #9DD0E7 0
      );
}

.emotion-3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  bottom: 4px;
  left: -1px;
  height: 12px;
}

<blockquote>
    <p>
      There are several ways to connect to your MongoDB instance.
    </p>
    <ol>
      <li>
         for access through a downloadable user interface
      </li>
      <li>
         interactive shell
      </li>
      <li>
        <span
          class="emotion-0 emotion-1"
          data-leafygreen-ui="anchor-container"
        >
          <span
            class="emotion-2"
          >
            programmatic access
          </span>
          <svg
            alt=""
            aria-hidden="true"
            class="emotion-3"
            height="16"
            role="presentation"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M13.823 2.4491C13.8201 2.30008 13.6999 2.17994 13.5509 2.17704L9.5062 2.09836C9.25654 2.09351 9.12821 2.39519 9.30482 2.5718L10.3856 3.65257L7.93433 6.10383C7.87964 6.15852 7.83047 6.21665 7.78683 6.27752L5.99909 8.06525C5.46457 8.59977 5.46457 9.4664 5.99909 10.0009C6.53361 10.5354 7.40023 10.5354 7.93475 10.0009L9.72249 8.21317C9.78336 8.16953 9.84148 8.12037 9.89618 8.06567L12.3474 5.61441L13.4282 6.69518C13.6048 6.87179 13.9065 6.74347 13.9016 6.4938L13.823 2.4491Z"
              fill="currentColor"
            />
            <path
              d="M7.25 3.12893C7.66421 3.12893 8 3.46472 8 3.87893C8 4.29315 7.66421 4.62893 7.25 4.62893H4C3.72386 4.62893 3.5 4.85279 3.5 5.12893V11.9929C3.5 12.2691 3.72386 12.4929 4 12.4929H10.864C11.1401 12.4929 11.364 12.2691 11.364 11.9929V8.75C11.364 8.33579 11.6998 8 12.114 8C12.5282 8 12.864 8.33579 12.864 8.75V11.9929C12.864 13.0975 11.9686 13.9929 10.864 13.9929H4C2.89543 13.9929 2 13.0975 2 11.9929V5.12893C2 4.02436 2.89543 3.12893 4 3.12893H7.25Z"
              fill="currentColor"
            />
          </svg>
        </span>
        
through a number of programming APIs.
      </li>
    </ol>
  </blockquote>
</DocumentFragment>
`;
